src/frontend/src/views/Upload/components/Samples/utils.ts

```diff
@@ -129,27 +129,49 @@ function handleFastaText(text: string, filename: string): ParseOutcome {
       continue;
     }
 
-    // Aggregate sequence line by line
-    for (let j = i + 1; j < lines.length; j++) {
-      const jLine = lines[j];
+    const { newIndex, sequence } = aggregateSequence(i, lines);
 
-      if (!jLine || jLine.includes(">")) {
-        i = j;
-        break;
-      }
+    i = newIndex;
 
+    if (sequence) {
       const id = iLine.replace(">", "");
 
       result = {
         ...result,
-        [id]: (result[id] || "") + jLine,
+        [id]: sequence,
       };
     }
   }
 
   return { errors, result };
 }
 
+function aggregateSequence(
+  i: number,
+  lines: string[]
+): { newIndex: number; sequence: string } {
+  let newIndex = i;
+  let sequence = "";
+
+  for (let j = i + 1; j < lines.length; j++) {
+    const jLine = lines[j];
+
+    if (!jLine || jLine.includes(">")) {
+      newIndex = j;
+      break;
+    }
+
+    sequence += jLine;
+
+    // (thuang): j has reached the end of file
```

Comment:
this is the only additional check in the algo. Once `j` reaches the end of file, we need to move `i` to the same index as well, otherwise the loop won't end because it never detects an empty line as available in some fasta files (we check the empty line at line `159`)